### PR TITLE
Re-connecting WPCOM Account - Adjust the linked accounts section in settings

### PR DESCRIPTION
**Pull request by eason9487**
### Changes proposed in this Pull Request:

It's a part of frontend in #775

- Move `<WPComAccountCard>`, `<GoogleMCAccountCard>`, and `<GoogleAdsAccountCard>` to the shared component folder.
   - 💡  Moved a lot of files, adjusted several files, and internalized WPCOM appearance to `<AccountCard>`. This is to give a little better consistency to their composition of components across all similar account cards.
- Change the linked accounts section in settings to use Card UI to present each account.

#### 📷  Not yet connected to a Google Ads account
image

#### 📷  Has a connected Google Ads account
image

### Detailed test instructions:

1. Go to the Settings page.
2. Check if the UI of **Linked Accounts** section is the same as the design in Figma.
3. Go to the onboarding setup and Google ads account setup pages.
4. Check if all account cards are displayed as the same as before.

### Additional details:

Figma link: eQ9O4m2flzrcAiDMXkOW0m-fi-6257%3A116720

### Changelog entry

> Tweak - Adjust the UI of the linked accounts section on the Settings page.

### CODE REVIEW

File: js/src/components/account-card/index.js [hunk 1]
```diff
@@ -38,11 +40,24 @@ const googleLogo = (
...
 const appearanceDict = {
 	[ APPEARANCE.EMPTY ]: {},
+	[ APPEARANCE.WPCOM ]: {
+		icon: wpLogo,
+		title: 'WordPress.com',
+	},
 	[ APPEARANCE.GOOGLE ]: {
 		icon: googleLogo,
-		title: __( 'Google account', 'google-listings-and-ads' ),
+		title: __( 'Google', 'google-listings-and-ads' ),
```
Comment (eason9487) [inline js/src/components/account-card/index.js:60]: 📝 
I checked several recent designs on Figma. All Google account cards no longer have "account" in the title, so I removed it here.

File: js/src/components/google-mc-account-card/connected-card.js [hunk 2]
```diff
@@ -18,8 +18,15 @@ import useDispatchCoreNotices from '.~/hooks/useDispatchCoreNotices';
...
 import { useAppDispatch } from '.~/data';
 
-const ConnectedCard = ( props ) => {
-	const { googleMCAccount } = props;
+/**
+ * Renders a Google Merchant Center account card UI with connected account information.
+ * It also provides a switch button that lets user connect with another account.
+ *
+ * @param {Object} props React props.
+ * @param {{ id: number }} props.googleMCAccount A data payload object contains user's Google Merchant Center account ID.
```
Comment (puntope) [inline js/src/components/google-mc-account-card/connected-card.js]:
```suggestion
 * @param {{ id: number }} props.googleMCAccount A data payload object containing the user's Google Merchant Center account ID.
```

File: js/src/components/google-ads-account-card/connected.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (puntope) [inline js/src/components/google-ads-account-card/connected.js]: 💅 / ❓  Why in `js/src/components/wpcom-account-card/index.js` this is named `connected-wpcom-account-card.js` whereas here is just `connected.js`?

**Comment by puntope** [comment]
💅  Not related to this PR but after seeing your nice work in refactoring the cards to the components I was wondering if it would be even nicer to organize the components folder. 

- Cards -> All the cards
- Icons -> Icons
- Controls -> Inputs / user interaction elements like inputs, selects, etc 
- Rest in the root...

**Comment by puntope** [comment]
❓ / 🚧  Unsure if this is an issue or it should be like this, but Figma looks different.

- Figma: No underline. (I guess it underlines on hover) 
- PR: Underlined, with kinda outline effect on hover


https://user-images.githubusercontent.com/5908855/161042505-fbebc108-a38b-42e5-9a7b-85e11326c813.mov

**Comment by puntope** [comment]
💅 / ❓ /  🚧  

Probably a heritage from previous implementations but the padding for the footer area in the Ads Cards is smaller in the Implementation than the Figma

Screenshot 2022-03-31 at 13 18 14

I guess in `js/src/wcdl/section/card/footer/index.scss` instead of `--large-gap / 2 ` would be more accurate `--large-gap / 1.5`

**Review by puntope — COMMENTED** [review]
✅  Overall looking good @eason9487 

Left a few comments 👌

File: js/src/components/google-mc-account-card/connected-card.js (hunk 2 above)
Comment (eason9487) [reply js/src/components/google-mc-account-card/connected-card.js]: Is there a grammar problem with this?

File: js/src/components/google-ads-account-card/connected.js (hunk 3 above)
Comment (eason9487) [reply js/src/components/google-ads-account-card/connected.js]: I originally intended to minimize the number of renaming files, so I used their original names. Considering that there are two exposed components, which may not match when globally searching for the codebase, I changed their internal names to the same one by https://github.com/woocommerce/google-listings-and-ads/pull/1374/commits/c0f7fe8fbfc6a681816e8af945c49fab6514aede.

**Comment by eason9487** [comment]
@puntope, thanks for the review!

> ❓ / 🚧 Unsure if this is an issue or it should be like this, but Figma looks different.
> 
> * Figma: No underline. (I guess it underlines on hover)
> * PR: Underlined, with kinda outline effect on hover
Kapture 2022-03-31 at 20 41 12

I cannot reproduce this problem. GLA should have a higher priority than `@wordpress/components` for button style overriding as the following SCSS styles. Probably it needs your help with checking the override relationship between them from DevTool.
- https://github.com/woocommerce/google-listings-and-ads/blob/af2cb919d1127f75313008ac7bdeee4fae1dcca8/js/src/css/shared/_gutenberg-components.scss#L2
- https://github.com/woocommerce/google-listings-and-ads/blob/af2cb919d1127f75313008ac7bdeee4fae1dcca8/js/src/css/shared/_gutenberg-components.scss#L29-L31

> 💅 / ❓ / 🚧
> 
> Probably a heritage from previous implementations but the padding for the footer area in the Ads Cards is smaller in the Implementation than the Figma

Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/1374/commits/af2cb919d1127f75313008ac7bdeee4fae1dcca8.

> 💅 Not related to this PR but after seeing your nice work in refactoring the cards to the components I was wondering if it would be even nicer to organize the components folder.

Sounds good! I added your comment to related issue #195.

File: js/src/components/google-mc-account-card/connected-card.js (hunk 2 above)
Comment (puntope) [reply js/src/components/google-mc-account-card/connected-card.js]: - props.googleMCAccount - A data payload object contains user's Google Merchant Center account ID. 

First is referring to googleMCAccount , but then Is saying That a Data Payload object, in general, contains Google Merchant Center Account ID. 

Could be like "props.fishEater - A person eats fish" 

- props.googleMCAccount -  A data payload object containing the user's Google Merchant Center account ID.

It is clear that we are actually referring to googleMCAccount being the payload object containing Google Merchant Center account ID. Also "A data payload that contains the user's Google Merchant Center account ID." works fine. 

Could be like "props.fishEater - A person that eats fish",  "props.fishEater - A person eating fish"  

It was definitely a super minor correction and I understood perfectly the intent of the property. But I as a non-native English speaker also love when someone points out my very usual English mistakes. Definitely I'm not the best to talk about English 🙈

**Comment by puntope** [comment]
> I cannot reproduce this problem. GLA should have a higher priority than `@wordpress/components` for button style overriding as the following SCSS styles. Probably it needs your help with checking the override relationship between them from DevTool.
> 
> *

Crazy, it comes from Automatewoo

Screenshot 2022-03-31 at 15 52 46

Screenshot 2022-03-31 at 15 55 04

**Comment by puntope** [comment]
> Probably a heritage from previous implementations but the padding for the footer area in the Ads Cards is smaller in the Implementation than the Figma

> Adjusted in https://github.com/woocommerce/google-listings-and-ads/commit/af2cb919d1127f75313008ac7bdeee4fae1dcca8.

I was meant the padding top/bottom that makes the footer are higher. Sorry I maybe was not clear or explaining myself in a confusing way.

Screenshot 2022-03-31 at 15 58 00

I guess in js/src/wcdl/section/card/footer/index.scss instead of `--large-gap / 2` would be more accurate to use --> `--large-gap / 1.5`

**Comment by puntope** [comment]
> Crazy, it comes from Automatewoo

As a suggestion, or potential solution maybe we can wrap the shared styles inside `.gla-something` to get priority in the cascade

File: js/src/components/google-mc-account-card/connected-card.js (hunk 2 above)
Comment (eason9487) [reply js/src/components/google-mc-account-card/connected-card.js]: Thanks for the explanation! I fixed them in https://github.com/woocommerce/google-listings-and-ads/pull/1374/commits/cb541963947610af17db28c699ad8799b15fd383.

**Comment by eason9487** [comment]
@puntope
 
> I was meant the padding top/bottom that makes the footer are higher. Sorry I maybe was not clear or explaining myself in a confusing way.

Oh sorry, it's my fault. I only looked at the screenshot yesterday and started fixing it. 🤦‍♂️ 
 
> I guess in js/src/wcdl/section/card/footer/index.scss instead of `--large-gap / 2` would be more accurate to use --> `--large-gap / 1.5`

The link style button has padding in Figma, and it has more legacy style issues between DEWP, `@wordpress/components`, and GLA. 



Therefore, I tried another way to fix them in \<AccountCard\> scope by https://github.com/woocommerce/google-listings-and-ads/pull/1374/commits/27489370b78e110ce8b9e714fce342a0dda07b8b.

With [another style fixing](https://github.com/woocommerce/google-listings-and-ads/blob/e928c8c2cc35206457be11f122bf916eda820df5/js/src/components/account-card/index.scss#L1-L2) in #1385, the card layout should be much closer to visual design. After all, there is still a 3px gap from the border style, but this will need to further override the default card style, so I guess the difference is probably tolerable.

---

> > Crazy, it comes from Automatewoo
> 
> As a suggestion, or potential solution maybe we can wrap the shared styles inside `.gla-something` to get priority in the cascade

That's a good point! Adding another `.gla-something` can set a higher priority, and this also makes sure GLA won't override other styles that are out of GLA pages. Since this's a bit out of this PR scope, I created a follow-up issue #1391 for it.

**Review by puntope — APPROVED** [review]
✅  Thanks for the changes and for creative ing the follow-up PR this is good to go

**Pull request merged**